EpsilonAuxBMS/Src/SetContactorsTask.c

```diff
@@ -109,15 +108,34 @@ void setContactorsTask(void const* arg)
 
                 if (!contactorError)
                 {
-                    state = CHARGE_CONTACTOR_ENABLE_CHECK;
-                    // Turn on charge Contactor
-                    HAL_GPIO_WritePin(CHARGE_CONTACTOR_ENABLE_GPIO_Port, CHARGE_CONTACTOR_ENABLE_Pin, GPIO_PIN_SET);
+                    state = CHARGE_AND_DISCHARGE_CONTACTOR_ENABLE_CHECK;
+                    charge = !HAL_GPIO_ReadPin(CHARGE_SENSE_GPIO_Port, CHARGE_SENSE_Pin);
+                    discharge = !HAL_GPIO_ReadPin(DISCHARGE_SENSE_GPIO_Port, DISCHARGE_SENSE_Pin);
+
+                    // Turn on charge and discharge contactors with a delay between them if neither of them
+                    // were on. There is no delay if one of them was on. This is done so there isn't any wasted
+                    // time
+                    if (!charge)
+                    {
+                        HAL_GPIO_WritePin(CHARGE_CONTACTOR_ENABLE_GPIO_Port, CHARGE_CONTACTOR_ENABLE_Pin, GPIO_PIN_SET);
+                    }
+
+                    if (!charge && !discharge)
+                    {
+                        osDelay(AUX_SET_CONTACTOR_FREQ);
+                        HAL_GPIO_WritePin(DISCHARGE_CONTACTOR_ENABLE_GPIO_Port, DISCHARGE_CONTACTOR_ENABLE_Pin, GPIO_PIN_SET);
+                    }
+                    else if (!discharge)
+                    {
+                        HAL_GPIO_WritePin(DISCHARGE_CONTACTOR_ENABLE_GPIO_Port, DISCHARGE_CONTACTOR_ENABLE_Pin, GPIO_PIN_SET);
+                    }
                 }
 
                 break;
 
-            case CHARGE_CONTACTOR_ENABLE_CHECK:
-                contactorError = !isContactorSet(CHARGE_SENSE_Pin, CHARGE_SENSE_GPIO_Port, 2);
+            case CHARGE_AND_DISCHARGE_CONTACTOR_ENABLE_CHECK:
+                contactorError = !isContactorSet(CHARGE_SENSE_Pin, CHARGE_SENSE_GPIO_Port, 3) ||
```

Comment:
By the time we get to this state, we're expecting that both of them should be on since 2 seconds would have passed, so the multiplier has to be 3 for normal behaviour.